### PR TITLE
Fixed autosave from triggering on empty forms

### PR DESCRIPTION
**Pull request by qhanson55**
# Description

When changing steps in the submission intake form autosave is no longer triggered, onBlur for phone number doesn't trigger autosave anymore, and no more double autosave on initial save of draft.
Keeps empty drafts from cluttering the draft and submissions table.

[PADS-337](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-337)

## Types of changes

Bug fix (non-breaking change which fixes an issue) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/file/AdvancedFileUpload.vue [hunk 1]
```diff
@@ -23,17 +23,37 @@ const {
   disabled?: boolean;
 }>();
 
+const emit = defineEmits<{
+  (event: 'activity-generated', activityId: string, submissionId: string): void;
+}>();
```
Comment (wilwong89) [inline frontend/src/components/file/AdvancedFileUpload.vue:28]: Emits in the script section should be in camel case for convention, so it should be `activityGenerated` both here and later in line 46. Vue will automatically detect it in kebab case later on in line 794 and trigger there even when it's `activity-generated` there.

File: frontend/src/components/form/internal/InputMaskInternal.vue [hunk 2]
```diff
@@ -14,6 +15,15 @@ const { label, name, mask, placeholder, disabled, bold } = defineProps<{
...
 const { errorMessage, value } = useField<string>(name);
+
+const normalizedValue = computed({
+  get: () => {
+    return value.value === '' ? undefined : value.value;
+  },
+  set: (val) => {
+    value.value = val === '' || val === undefined ? (undefined as unknown as string) : val;
+  }
+});
```
Comment (wilwong89) [inline frontend/src/components/form/internal/InputMaskInternal.vue:26]: Have you checked to see if this might break form submissions in other components? This is a widespread change to every view that uses an input field.

File: frontend/src/components/housing/submission/SubmissionIntakeForm.vue [hunk 3]
```diff
@@ -66,6 +66,7 @@ import type { AutoCompleteCompleteEvent } from 'primevue/autocomplete';
 import type { DropdownChangeEvent } from 'primevue/dropdown';
 import type { IInputEvent } from '@/interfaces';
 import type { Document, Permit, Submission } from '@/types';
+import type { FormContext } from 'vee-validate';
```
Comment (wilwong89) [inline frontend/src/components/housing/submission/SubmissionIntakeForm.vue:69]: Remove unused import.

**Comment by slhurley** [comment]
Looking almost perfect!  I did have one weird thing where I didn't fill out the Address section, clicked Submit and the validation occurred telling me I forgot that field...I stopped to read the red error message and it all of a sudden auto-saved and the validation went away...but I couldn't make it happen again!  But I did find one error...go back and fill out the address/coordinates and click Submit and I get an Axios Error 422 (Failed to save intake) after clicking Yes to submitting so I couldn't actually submit my submission.

File: frontend/src/components/form/internal/InputMaskInternal.vue (hunk 2 above)
Comment (qhanson55) [reply frontend/src/components/form/internal/InputMaskInternal.vue:26]: It's just for the InputMask field not any other input fields, and it's only changing the it so that it handles no input as undefined rather than it's default of empty string, which triggers autosave, as empty string is technically a value. I tested other InputMasks and all seems fine, and test seem to be passing at same levels as before.

File: frontend/src/components/file/AdvancedFileUpload.vue [hunk 4]
```diff
@@ -23,17 +23,37 @@ const {
...
+      return response.data.activityId;
+    } else {
+      return undefined;
+    }
+  } catch (error) {
+    toast.error('Failed to generate activity ID');
+    return undefined;
+  }
+}
+
```
Comment (kyle1morel) [inline frontend/src/components/file/AdvancedFileUpload.vue:56]: This is a lot of logic unrelated to file uploading. Can this be moved to the intake form?

File: frontend/src/components/file/AdvancedFileUpload.vue [hunk 5]
```diff
@@ -55,12 +75,18 @@ const onFileUploadDragAndDrop = (event: FileUploadUploaderEvent) => {
 const onUpload = async (files: Array<File>) => {
   uploading.value = true;
 
+  if (!activityId) {
+    curActivityId.value = await generateActivityId();
+  } else {
+    curActivityId.value = activityId;
+  }
```
Comment (jujaga) [inline frontend/src/components/file/AdvancedFileUpload.vue:82]: Optional style: You might be able to get away with the following one-liner ternary
```ts
curActivityId.value = activityId ? activityId : await generateActivityId();
```

File: frontend/src/components/file/AdvancedFileUpload.vue (hunk 4 above)
Comment (qhanson55) [reply frontend/src/components/file/AdvancedFileUpload.vue]: done

File: frontend/src/components/file/AdvancedFileUpload.vue (hunk 5 above)
Comment (qhanson55) [reply frontend/src/components/file/AdvancedFileUpload.vue]: done and noted!

**Review by wilwong89 — APPROVED** [review]

**Review by kyle1morel — APPROVED** [review]
Approving as is. Changes will have to be made to this due to the new draft saving functionality happening in another branch.
